contracts/scripts/test.sh

```diff
@@ -8,28 +8,59 @@ function play {
 
   local contract_name=$1
   local contract_address=$(cat "$CONTRACTS_PATH"/addresses.json | jq --raw-output ".$contract_name")
+  local ticket_address=$(cat "$CONTRACTS_PATH"/addresses.json | jq --raw-output ".${contract_name}_ticket")
 
-  cd "$CONTRACTS_PATH"/$contract_name
+  # airdrop initial tickets
 
-  echo "calling press for" $contract_name "["$contract_address"]" "by" $PLAYER1_SEED
+  cd "$CONTRACTS_PATH"/ticket_token
 
-  cargo contract call --url $NODE --contract $contract_address --message IButtonGame::press --suri $PLAYER1_SEED
+  echo "sending ticket token" ${contract_name}_ticket "["$ticket_address"]" "to " $PLAYER1
 
-  sleep 1
+  cargo contract call --url $NODE --contract $ticket_address --message PSP22::transfer --args $PLAYER1 1 "[0]" --suri $AUTHORITY_SEED
 
-  echo "calling press for" $contract_name "["$contract_address "]" "by" $PLAYER2_SEED
+  echo "sending ticket token" ${contract_name}_ticket "["$ticket_address"]" "to " $PLAYER2
 
-  cargo contract call --url $NODE --contract $contract_address --message IButtonGame::press --suri $PLAYER2_SEED
+  cargo contract call --url $NODE --contract $ticket_address --message PSP22::transfer --args $PLAYER2 1 "[0]" --suri $AUTHORITY_SEED
 
-  # --- TRIGGER DEATH AND REWARDS DISTRIBUTION
+  # give allowance for spending tickets to the game contract
 
-  sleep $(($LIFETIME + 1))
+  cd "$CONTRACTS_PATH"/ticket_token
 
-  echo "claiming rewards for" $contract_name
-  cargo contract call --url $NODE --contract $contract_address --message IButtonGame::claim_reward --args $PLAYER1 --suri $AUTHORITY_SEED
-  cargo contract call --url $NODE --contract $contract_address --message IButtonGame::claim_reward --args $PLAYER2 --suri $AUTHORITY_SEED
+  echo "allowing" $contract_name "["$contract_address"]" "to spend up to" $TOTAL_BALANCE "of" ${contract_name}_ticket "["$ticket_address"]" "on behalf of" $PLAYER1
 
-  echo "Done playing" $contract_name
+  cargo contract call --url $NODE --contract $ticket_address --message PSP22::approve --args $contract_address $TOTAL_BALANCE --suri $PLAYER1_SEED
+
+  echo "allowing" $contract_name "["$contract_address"]" "to spend up to" $TOTAL_BALANCE "of" ${contract_name}_ticket "["$ticket_address"]" "on behalf of" $PLAYER2
+
+  cargo contract call --url $NODE --contract $ticket_address --message PSP22::approve --args $contract_address $TOTAL_BALANCE --suri $PLAYER2_SEED
+
+  # TODO : can't test before mint / burn for game tokens is implemented; uncomment when A0-1236 is done
```

Comment:
Comment vs. issue as before.

Reply:
There is an existing issue for that: https://cardinal-cryptography.atlassian.net/browse/A0-1236